File: backend/app/models/notification.py
import uuid
from sqlalchemy import Column, String, DateTime, ForeignKey, Boolean
from sqlalchemy.sql import func

from app.database import Base


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id"), nullable=False)
    complaint_id = Column(String(36), ForeignKey("complaints.id"), nullable=True)

    title = Column(String, nullable=False)
    message = Column(String, nullable=False)
    notification_type = Column(String, nullable=False)
    is_read = Column(Boolean, default=False, nullable=False)

    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: frontend/types/notification.ts
export interface Notification {
  id: string
  user_id?: string
  complaint_id?: string | null
  title: string
  message: string
  notification_type?: string
  is_read: boolean
  created_at?: string
}

export interface NotificationEvent {
  type: string
  payload: Notification
}
